### ISSUE TITLE
catch EINTR in select and retry?

### ISSUE THREAD

**Issue by msprotz** [dsheets/ocaml-process#9]
Hello David,

I suddenly started having a series of failures on my travis build. After a little bit of diagnosing, here's what I see:

https://travis-ci.org/FStarLang/kremlin/builds/282423569?utm_source=slack&utm_medium=notification

there's a bunch of `Fatal error: exception Unix.Unix_error(Unix.EINTR, "select", "")` that are thrown from within `process.ml`; is it the case that calling `select` should be retried, just like for `waitpid`?

Thanks,

Jonathan

**Comment by msprotz** [comment]
Sadly, I restarted the build and after three consecutive failures it started to work. I did enable backtraces, though, and saw that the exception was thrown within process.ml -- looks like travis doesn't have a way of seeing the logs for a previous attempt at the same build, though.

**Comment by dsheets** [comment]
Yes, `EINTR` should probably be checked and `select` retried. I'd be happy to accept a PR or I can probably do it in a few days.

**Comment by msprotz** [comment]
A few days is fine -- the error seems to be gone for now... thanks!
